rpc/chain/client.go

```diff
@@ -25,6 +25,10 @@ type FeeHistory struct {
 	BaseFeePerGas []string `json:"baseFeePerGas"`
 }
 
+type EthClient interface {
```

Comment:
This is the Client abstract interface that `ClientWithFallback` implements. It is to be extended with mocked values used in tests. Not sure if the fact that we now have only one function abstraction makes sense to name it as such. Still, its scope is to abstract the `ethclient.Client`. Maybe you will like `ClientInteface` better?

Reply:
Because the `Client` object is one of our main wallet dependencies and the most important one I think it should have been named `ClientInterface` or `ClientIf` from the start and used everywhere instead of `go-ethereum.Client` so it can be easily mocked to test the client. However, I think it is confusing to have intermediate names just because we do it incrementally, now for one call and extend later with other calls as needed.